src/Endpoints/Keys.php

```diff
@@ -65,18 +65,32 @@ protected function fill(array $attributes): self
         $this->actions = $attributes['actions'];
         $this->indexes = $attributes['indexes'];
         if ($attributes['expiresAt']) {
-            $this->expiresAt = date_create_from_format('Y-m-d\TH:i:s\Z', $attributes['expiresAt']);
+            $this->expiresAt = $this->createDate($attributes['expiresAt']);
         }
         if ($attributes['createdAt']) {
-            $this->createdAt = date_create_from_format('Y-m-d\TH:i:s.vu\Z', $attributes['createdAt']);
+            $this->createdAt = $this->createDate($attributes['createdAt']);
         }
         if ($attributes['updatedAt']) {
-            $this->updatedAt = date_create_from_format('Y-m-d\TH:i:s.vu\Z', $attributes['updatedAt']);
+            $this->updatedAt = $this->createDate($attributes['updatedAt']);
         }
 
         return $this;
     }
 
+    protected function createDate($attribute): ?DateTime
+    {
+        if (!\is_string($attribute)) {
+            return null;
+        }
+
+        $date = date_create_from_format('Y-m-d\TH:i:s.uP', $attribute);
```

Comment:
> date_create_from_format
> (PHP 5 >= 5.3.0, PHP 7, PHP 8)
> 
> date_create_from_format — Alias of [DateTime::createFromFormat()]
(https://www.php.net/manual/en/datetime.createfromformat.php)

> 
> Description[ ¶](https://www.php.net/manual/en/function.date-create-from-format.php#refsect1-function.date-create-from-format-description)
> This function is an alias of: [DateTime::createFromFormat()](https://www.php.net/manual/en/datetime.createfromformat.php)

I suggest using the` DateTime::createFromFormat `or even better `DateTimeImmutable::createFromFormat` as these timestamps are not expected to be modified.

Reply:
I think this should be a separate improvement if so. This has sat around long enough.